lms/services/hapi.py

```diff
@@ -133,8 +134,12 @@ def request(
         except RequestException as err:
             response = getattr(err, "response", None)
             status_code = getattr(response, "status_code", None)
+            if status_code == 404:
+                exception_class = HAPINotFoundError
+            else:
+                exception_class = HAPIError
```

Comment:
The views need to distinguish between a 404 from h and any other kind of error, so the API client now raises a different exception type for 404s.

Reply:
I have a little concern about how this will scale if we need to switch on other status codes in the future, but given the state of things overall, I think it's fine for now.

Reply:
I think it should scale fine. If the `if` / `else` gets too long we can replace it with a dict mapping status codes to exception classes. How many different kinds of error response, that the LMS app is interested in special-casing, are there ever likely to be in the future? I don't think it'll grow too out of hand. And having a different exception subclass for each status code makes the code that uses this and its test simplest